src/index.js

```diff
@@ -41,41 +43,66 @@ const printList = () => {
+    const newDescription = document.getElementById(`edit${taskIndex}`).value;
+    const taskElement = document.getElementById(`${taskIndex}`);
+    const task = new Task(newDescription, taskIndex);
+    task.Edit();
+
+    document.querySelector(`#edit${taskIndex}`).classList.add('hidden');
+    document.querySelector(`#task${taskIndex}`).classList.remove('hidden');
+
+    taskElement.querySelector('.edit-manager').classList.add('hidden');
+    taskElement.querySelector('.list-editor').classList.remove('hidden');
+    printList();
+  };
+
+  // Start editting
   document.querySelectorAll('.edit-task').forEach((etb) => {
     etb.addEventListener('click', (e) => {
       const taskIndex = parseInt(e.target.parentNode.parentNode.id, 10);
+      document.getElementById(`edit${taskIndex}`).value = document.querySelector(`#task${taskIndex}`).innerHTML;
       const taskElement = document.getElementById(`${taskIndex}`);
-      // console.log(taskElement);
-
-      document.querySelector(`#edit${taskIndex}`).classList.remove('hidden');
+      const editTask = document.querySelector(`#edit${taskIndex}`);
+      editTask.classList.remove('hidden');
       document.querySelector(`#task${taskIndex}`).classList.add('hidden');
 
       taskElement.querySelector('.edit-manager').classList.remove('hidden');
       taskElement.querySelector('.list-editor').classList.add('hidden');
 
-      // console.log(document.querySelector(`#task${taskIndex}`));
-      // printList();
+      editTask.addEventListener('keypress', (e) => {
+        const keypressed = (editTask) ? e.keyCode : e.which;
+        if (keypressed === 13) {
+          if (editTask.value !== '') { confirmEditTask(e); }
+        }
+      });
     });
   });
+
+  // Confirm task edition
   document.querySelectorAll('.edit-confirm').forEach((ec) => {
+    ec.addEventListener('click', confirmEditTask);
+  });
+
+  // Cancel task etition
+  document.querySelectorAll('.edit-cancel').forEach((ec) => {
     ec.addEventListener('click', (e) => {
       const taskIndex = parseInt(e.target.parentNode.parentNode.id, 10);
-      const newDescription = document.getElementById(`edit${taskIndex}`).value;
       const taskElement = document.getElementById(`${taskIndex}`);
-      const task = new Task(newDescription, taskIndex);
-      task.Edit();
-      // console.log(newDescription);
 
       document.querySelector(`#edit${taskIndex}`).classList.add('hidden');
       document.querySelector(`#task${taskIndex}`).classList.remove('hidden');
 
       taskElement.querySelector('.edit-manager').classList.add('hidden');
       taskElement.querySelector('.list-editor').classList.remove('hidden');
 
-      // console.log(document.querySelector(`#task${taskIndex}`));
       printList();
     });
   });
+
+  // delete selected task
   document.querySelectorAll('.delete-task').forEach((dtb) => {
     dtb.addEventListener('click', (e) => {
       const task = new Task();
```

Comment:
[OPTIONAL]
- [ ] Deleting also isn't working for some reason, here is the error:
  ![image](https://user-images.githubusercontent.com/20567503/196301107-17f0a581-acde-409a-be3f-dac20df6bd64.png)